### ISSUE TITLE
Severe network bug on v1.15.4

### ISSUE THREAD

**Issue by buroa** [cilium/cilium#32095]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Upon upgrading from v1.15.3 -> v1.15.4 and doing a few node reboots, some nodes start failing pod connectivity.

```
Controller Status:      123/124 healthy
  Name                                  Last success   Last error   Count   Message
  endpoint-1578-regeneration-recovery   never          38s ago      39      regeneration recovery failed
  ```

It seems BPF gets in a really weird state trying to come back and then I see this as well: `creating bpffs link dir for device bond0.713: mkdir /sys/fs/bpf/cilium/devices/bond0.713: operation not permitted`

Once I roll back to v1.15.3, it creates the endpoint and pod connectivity comes back immediately.

### Cilium Version

v1.15.4

### Kernel Version

6.1.82

### Kubernetes Version

v1.29.4

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aditighag** [comment]
Please attach a sysdump from the affected cluster.

**Comment by log1cb0mb** [comment]
Just ran into this as well.

**Comment by axeII** [comment]
Yesterday, I attempted to upgrade to Cilium 1.15.4. The upgrade failed on some nodes and after that some nodes stoped properly working. Rebooting them resolved the issue. However, this morning, I received an alert from Flux about Cilium reconciliation failure. Reverting back to version 1.15.3 resolved all issues. 

I will try to dig some logs if possible.

**Comment by buroa** [comment]
@aditighag https://pub-d231a33d470140839d37b46626308bfa.r2.dev/cilium-sysdump-20240421-072009.zip

**Comment by buroa** [comment]
```Errors:                cilium             cilium-bvccn    controller endpoint-578-regeneration-recovery is failing since 13s (18x): regeneration recovery failed
                       cilium             cilium-94wz4    controller endpoint-322-regeneration-recovery is failing since 29s (17x): regeneration recovery failed
                       cilium             cilium-6lhkh    controller endpoint-1516-regeneration-recovery is failing since 29s (17x): regeneration recovery failed
                       cilium             cilium-xpxcx    controller endpoint-1725-regeneration-recovery is failing since 13s (18x): regeneration recovery failed
Warnings:              cilium             cilium-xpxcx    1 endpoints are not ready
                       cilium             cilium-z8x9f    1 endpoints are not ready
                       cilium             cilium-bvccn    1 endpoints are not ready
                       cilium             cilium-94wz4    1 endpoints are not ready
                       cilium             cilium-6lhkh    1 endpoints are not ready
                       cilium             cilium-vj9rx    1 endpoints are not ready
                       ```

`kubectl exec --stdin --tty -n kube-system cilium-xpxcx -- cilium endpoint get 1725`
...
```      "log": [
        {
          "code": "Failure",
          "message": "Error regenerating endpoint: creating bpffs link dir for device eth0.10: mkdir /sys/fs/bpf/cilium/devices/eth0.10: operation not permitted",
          "state": "ready",
          "timestamp": "2024-04-21T12:25:15Z"
        }
      ],
```

**Comment by julianwiedmann** [comment]
Backport of https://github.com/cilium/cilium/pull/32090 is landing via https://github.com/cilium/cilium/pull/32103.

**Comment by bck01215** [comment]
It seems to me `mkdir /sys/fs/bpf/cilium/devices/bond0.713` the dot in the bod name causes the issue. Running that command on the server fails with permission denied but works if you remove the dot.

**Comment by julianwiedmann** [comment]
> Backport of #32090 is landing via #32103.

Merged, will come in the next v1.15 release.

**Comment by samip5** [comment]
How does one fix it before the next v1.15 release ships? Just use older v1.15?
